packages/meditrak-server/src/apiV2/surveyResponse.js

```diff
@@ -37,14 +37,16 @@ async function validateResponse(models, userId, body) {
   const surveyResponseValidator = createSurveyResponseValidator(models);
   await surveyResponseValidator.validate(body);
 
+  const survey = await models.survey.findById(body.survey_id);
+
   if (!body.entity_id && !body.entity_code) {
     throw new Error('Must provide one of entity_id or entity_code');
   }
 
   const surveyQuestions = await findQuestionsInSurvey(models, body.survey_id);
 
   const { answers } = body;
-  if (Object.keys(answers).length === 0) {
+  if (survey.can_repeat && Object.keys(answers).length === 0) {
```

Comment:
Honestly I do still feel this is acceptable even if the survey isn't periodic? @edmofro mind if I ask for your opinion on this?

The way I see it, it's a valid survey response even if there's no answers with it. It mightn't mean much, but it doesn't seem invalid..?

Reply:
Removed the answers validation altogether, making it allowed for all types of surveys.